### PR TITLE
Mig mgmtsystem review to v. 10.0

### PR DESCRIPTION
**Pull request by eugen-don**
Mig mgmtsystem review to v. 10.0
depends: 
- [x] https://github.com/OCA/management-system/pull/169 mgmtsystem nonconformity
- [x] https://github.com/OCA/management-system/pull/170 mgmtsystem_survey

### CODE REVIEW

File: mgmtsystem_review/models/__init__.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # -*- coding: utf-8 -*-
 
-from. import mgmtsystem_review_line
-from . import mgmtsystem_review
+from . import (
+    mgmtsystem_review,
+    mgmtsystem_review_line,
+)
```
Comment (dreispt) [inline mgmtsystem_review/models/__init__.py]: The previous format if OK (and more concise). No point in changing it.

File: mgmtsystem_review/tests/test_create_review.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
 
-from openerp.tests import common
+from odoo.tests import common
 from datetime import datetime
```
Comment (angelmoya) [inline mgmtsystem_review/tests/test_create_review.py:4]: Question for odoo experts: 

could be better to replace ?: 
```
from datetime import datetime
...
datetime.now()

```
to : 

```
from odoo import fields
...
fields.datetime.now()
```

**Comment by eugen-don** [comment]
Hi @max3903 @dreispt 
i had problems reverting or lets say rewriting the first commit message, luckily figured out how git reset command works.

Do you think this looks ok now?

**Comment by max3903** [comment]
@eugen-don Please look at the report attached:
[Review report (9).pdf](https://github.com/OCA/management-system/files/1071511/Review.report.9.pdf)

I can't see the title and the footer.

**Review by max3903 — CHANGES_REQUESTED** [review]
Report needs fixing.

**Comment by eugen-don** [comment]
@max3903  how did you print this report i cant find the menuitem?

**Comment by max3903** [comment]
@eugen-don I created a review and went to the print menu.

**Comment by eugen-don** [comment]
@max3903 

reports are OK, check it out one more time, paper format wasnt set accordingly.

**Review by max3903 — APPROVED** [review]

**Comment by eugen-don** [comment]
ping @angelmoya

**Comment by angelmoya** [comment]
Sorry @eugen-don this week I have a lot of work and no time to check this, maybe friday or next week.

Please @acm1pt-colorado check this

**Review by ljimenezsidoo — APPROVED** [review]
Hi @eugen-don tested on runbot, looks great

**Comment by angelmoya** [comment]
On runbot I create "AngelMoyaReview" and get this error when I try to print

```
Traceback (most recent call last):
  File "/home/odoo/odoo-10.0/addons/report/controllers/main.py", line 96, in report_download
    response = self.report_routes(reportname, docids=docids, converter='pdf')
  File "/.repo_requirements/odoo/odoo/http.py", line 504, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/odoo-10.0/addons/report/controllers/main.py", line 45, in report_routes
    pdf = report_obj.with_context(context).get_pdf(docids, reportname, data=data)
  File "/home/odoo/odoo-10.0/addons/report/models/report.py", line 178, in get_pdf
    html = self.with_context(context).get_html(docids, report_name, data=data)
  File "/home/odoo/odoo-10.0/addons/report/models/report.py", line 144, in get_html
    return self.render(report.report_name, docargs)
  File "/home/odoo/odoo-10.0/addons/report/models/report.py", line 120, in render
    return view_obj.render_template(template, values)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 1037, in render_template
    return self.browse(self.get_view_id(template)).render(values, engine)
  File "/home/odoo/odoo-10.0/addons/website/models/ir_ui_view.py", line 110, in render
    return super(View, self).render(values, engine=engine)
  File "/home/odoo/odoo-10.0/addons/web_editor/models/ir_ui_view.py", line 26, in render
    return super(IrUiView, self).render(values=values, engine=engine)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_ui_view.py", line 1057, in render
    return self.env[engine].render(self.id, qcontext)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_qweb/ir_qweb.py", line 53, in render
    return super(IrQWeb, self).render(id_or_xml_id, values=values, **context)
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 248, in render
    self.compile(template, options)(self, body.append, values or {})
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 317, in _compiled_fn
    raise e
QWebException: tuple index out of range
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 315, in _compiled_fn
    return compiled(self, append, values, options, log)
  File "<template>", line 1, in template_mgmtsystem_review_review_report_1735
  File "<template>", line 4, in body_call_content_1734
  File "<template>", line 74, in foreach_1722
  File "/.repo_requirements/odoo/odoo/models.py", line 5215, in __getitem__
    return self._browse((self._ids[key],), self.env)
IndexError: tuple index out of range

Error to render compiling AST
IndexError: tuple index out of range
Template: mgmtsystem_review.review_report
Path: /templates/t/t/div/div[4]/div[3]/t[1]/ul/li[1]/span
Node: <span t-field="response.user_input_line_ids[2].value_text"/>
```

**Comment by eugen-don** [comment]
@angelmoya 
The reson for the error is that the your review contains survey answers which arent completed yet. See in your review there are survey answers that are partly complete. Try to first complete the survey answers and then add the survey answers to the review again.

**Comment by angelmoya** [comment]
Nice @eugen-don , would be better to show this information in warning to prevent this error.

**Comment by eugen-don** [comment]
@angelmoya 
indeed thanks for the good idea. 

@dreispt @max3903 
This behaviour is the same in all versions.

I see two ways of correcting this:
1. add a filter to the Survey Answers field in the view definition to only show complete answers
2. edit the template to check if the surveys answer is completed and only print the answer if it is completed.

Since the first option is easier to code and i assume that including incomplete or empty answers in the modules data and reports provides no benefits, id like to stick to the first option.

What do you guys think?

**Comment by max3903** [comment]
@eugen-don Yes, first option makes more sense.

**Comment by eugen-don** [comment]
@dreispt 

could you merge?

**Comment by max3903** [comment]
@eugen-don Have you added the filter on the survey answer?

**Comment by eugen-don** [comment]
@max3903  no since its not part of the migration, its an improvement or fix and goes into a different PR

since the bug is present in versions 8.0 and 9.0 ill need to fix those first...

but i see no problems on adding a fix right in the PR of the migration of the module.

**Pull request merged**